File: Exercism/dart/raindrops/lib/raindrops.dart

class Raindrops {
  String convert(int number) {
    var result = StringBuffer();

    if (number % 3 == 0)
      result.write("Pling");

    if (number % 5 == 0)
      result.write("Plang");

    if (number % 7 == 0)
      result.write("Plong");

    if (result.isEmpty)
      result.write(number.toString());

    return result.toString();
  }
}

File: Exercism/typescript/raindrops/raindrops.ts

class Raindrops {
    convert(x: number): string {
        let result = "";

        if (x % 3 == 0)
            result += "Pling";

        if (x % 5 == 0)
            result += "Plang";

        if (x % 7 == 0)
            result += "Plong";

        if (result.length == 0)
            result += x.toString(10);

        return result;
    }
}

export default Raindrops
